harness/determined/cli/trial.py

```diff
@@ -175,11 +176,17 @@ def download(args: Namespace) -> None:
         checkpoints = det.get_trial(args.trial_id).list_checkpoints(
             sort_by=sort_by,
             order_by=order_by,
-            max_results=1,
         )
         if not checkpoints:
             raise ValueError(f"No checkpoints found for trial {args.trial_id}")
-        checkpoint = checkpoints[0]
+
+        valid_states = [CheckpointState.COMPLETED, CheckpointState.PARTIALLY_DELETED]
```

Comment:
So, something to think about... I don't love any decision, and I'll support whatever you decide:

It kind of sucks for `cli.trial.download` to have to know details about `Checkpoint.download` (namely, that `Checkpoint.download` only works for completed or partially deleted checkpoints). It kind of leaks implementation from the "checkpoint" abstraction.

A way around this is to just try/catch each `Checkpoint.download` / CheckpointStateException, no fewer details leaked.

But this isn't perfect because:
1. you still have to know that CheckpointStateExceptions are a thing
2. the implementation feels a little gross for reasons I can't quite verbalize

I think my own preference lies in this try/catch, but it's super easy for me to imagine people preferring the filter as written, and I think that's a fine choice, too.

If you also prefer it as written I support you in it. Just wanted to give you the prompt to think about it.

Reply:
I think the filtering implementation is clearer. We don't have to obfuscate why `download` fails in this particular case.